### PR TITLE
Use new errors

### PR DESCRIPTION
**Pull request by cgwalters**

### CODE REVIEW

File: glnx-fdio.c [hunk 1]
```diff
@@ -396,15 +371,8 @@ glnx_fd_readall_malloc (int               fd,
...
- out:
-  if (success)
-    {
-      *out_len = buf_size;
-      return buf;
-    }
-  g_free (buf);
-  return NULL;
+  *out_len = buf_size;
+  return g_steal_pointer (&buf);
```
Comment (cgwalters) [inline glnx-fdio.c:375]: In particular, this looks way better IMO.

**Comment by gnomesysadmins** [comment]
Thank you for contributing to libglnx!

libglnx uses Bugzilla for code review.

If you have never contributed to GNOME before make sure you have read the
getting started documentation:
http://www.gnome.org/get-involved

Otherwise please visit
https://wiki.gnome.org/Newcomers
and follow the instructions there to upload your change to Bugzilla.

File: glnx-fdio.c [hunk 2]
```diff
@@ -246,9 +237,10 @@ glnx_link_tmpfile_at (int dfd,
            */
           if (renameat (dfd, tmpfile_path, target_dfd, target) < 0)
             {
+              int errsv = errno;
               (void) unlinkat (dfd, tmpfile_path, 0);
-              glnx_set_error_from_errno (error);
-              return FALSE;
+              errno = errsv;
```
Comment (jlebon) [inline glnx-fdio.c:242]: Ouhhh, good catch.

**Review by jlebon — COMMENTED** [review]
👍 LGTM

**Comment by cgwalters** [comment]
Thanks,  merged.

**Pull request closed without merging**